recon/src/protocol.rs

```diff
@@ -429,6 +449,7 @@ where
         to_writer: &mut ToWriterSender<InitiatorMessage<R::Key, R::Hash>>,
         remote_range: RangeHash<R::Key, R::Hash>,
     ) -> Result<()> {
+        self.common.persist_all().await?;
```

Comment:
yeah, that seems reasonable. we have to persist before we do the range otherwise we don't get the same result for all the values we just received. I can clean up the comments/handling a bit for sure.

Reply:
Will address in #473.